lib/vcard/field.rb

```diff
@@ -241,13 +241,15 @@ def copy
       # fields that are already wrapped.
       def encode(width=nil)
         width = 75 unless width
-        l = @line
-        # Wrap to width, unless width is zero.
-        if width > 0
-          l = l.gsub(/.{#{width},#{width}}/) { |m| m + "\n " }
+        l = @line.rstrip
+        if width.zero?
+          l + "\n"
+        elsif width <= 1
+          raise ArgumentError, "#{width} is too narrow"
+        else
+          # Wrap to width
+          l.scan(/\A.{,#{width}}|.{1,#{width - 1}}/).join("\n ") + "\n"
```

Comment:
@brendon I believe this regexp is clear and easy to understand.  The pattern says the first line should be cut to the given number of characters in length and the following lines have a leading space so should be cut to the given number minus one in length.

When `width` is `3` and `l` is `"aaabbccd"`, then `l.scan(/\A.{,#{width}}|.{1,#{width - 1}}/)` will be evaluated to `["aaa", "bb", "cc", "d"]`, and applying`.join("\n ")` to it will result in `"aaa\n bb\n cc\n d"`.

Reply:
Lower bounds of 0 and 1 are not to get stuck when the input string is empty or multiples of the width plus one in length, respectively.

Reply:
Thanks @knu, that sounds good to me :)